File: src/HyperTrack/data_types/LocationError.ts
import type { HyperTrackError } from './HyperTrackError';
import type { NotRunning, Starting } from './internal/LocationErrorInternal';

export type Errors = {
  type: 'errors';
  value: HyperTrackError[];
};

export type LocationError = NotRunning | Starting | Errors;

File: android/src/main/java/com/reactnativehypertracksdk/common/LocationError.kt
package com.reactnativehypertracksdk.common

/**
 * Check HyperTrack SDK docs for LocationError
 */
internal sealed class LocationError
internal object NotRunning : LocationError()
internal object Starting : LocationError()
internal data class Errors(val errors: Set<HyperTrackError>) : LocationError()
